jwcrypto/jwk.py

```diff
@@ -482,6 +482,23 @@ class JWKSet(set):
 
     Inherits for the standard 'set' bultin type.
     """
+    def __init__(self, keyset=None):
+        if isinstance(keyset, str):
+            # Assume that this was a reimport
+            super(JWKSet, self).__init__()
+
+            try:
+                keyset = json_decode(keyset)['keys']
+            except:
+                raise InvalidJWKValue()
+
+            for jwk in keyset:
+                self.add(JWK(**jwk))
+
+        elif keyset is None:
+            super(JWKSet, self).__init__()
+        else:
+            super(JWKSet, self).__init__(keyset)
```

Comment:
This will call the set() constructor that takes an optional iterable argument to initialize it with.
Example: set([1,2,3]).
I didn't want to break anything in the API: I don't know if anyone would be using this constructor.